adapters/resetdigital/resetdigital.go

```diff
@@ -4,305 +4,237 @@ import (
-	Banner resetDigitalMediaType `json:"banner,omitempty"`
-	Video  resetDigitalMediaType `json:"video,omitempty"`
-	Audio  resetDigitalMediaType `json:"audio,omitempty"`
-}
-type resetDigitalMediaType struct {
-	Sizes [][]int64 `json:"sizes,omitempty"`
-	Mimes []string  `json:"mimes,omitempty"`
-}
-type resetDigitalBidResponse struct {
-	Bids []resetDigitalBid `json:"bids"`
+
+type adapter struct {
+	endpoint string
 }
-type resetDigitalBid struct {
-	BidID string  `json:"bid_id"`
-	ImpID string  `json:"imp_id"`
-	CPM   float64 `json:"cpm"`
-	CID   string  `json:"cid,omitempty"`
-	CrID  string  `json:"crid,omitempty"`
-	AdID  string  `json:"adid"`
-	W     string  `json:"w,omitempty"`
-	H     string  `json:"h,omitempty"`
-	Seat  string  `json:"seat"`
-	HTML  string  `json:"html"`
+
+func Builder(_ openrtb_ext.BidderName, cfg config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: cfg.Endpoint,
+	}, nil
 }
 
-func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
-	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
-	if err != nil {
-		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
-	}
-	bidder := &adapter{
-		endpoint: template,
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	if len(request.Imp) != 1 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "ResetDigital adapter supports only one impression per request",
+		}}
 	}
-	return bidder, nil
-}
 
-func getHeaders(request *openrtb2.BidRequest) http.Header {
-	headers := http.Header{}
-
-	addNonEmptyHeaders(&headers, map[string]string{
-		"Content-Type": "application/json;charset=utf-8",
-		"Accept":       "application/json",
-	})
-
-	if request != nil && request.Device != nil {
-		addNonEmptyHeaders(&headers, map[string]string{
-			"Accept-Language": request.Device.Language,
-			"User-Agent":      request.Device.UA,
-			"X-Forwarded-For": request.Device.IP,
-			"X-Real-Ip":       request.Device.IP,
-		})
+	imp := request.Imp[0]
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error parsing bidderExt from imp.ext: %v", err),
+		}}
 	}
-	if request != nil && request.Site != nil {
-		addNonEmptyHeaders(&headers, map[string]string{
-			"Referer": request.Site.Page,
-		})
+
+	var resetDigitalExt openrtb_ext.ImpExtResetDigital
+	if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error parsing resetDigitalExt from bidderExt.bidder: %v", err),
+		}}
 	}
 
-	return headers
-}
+	if resetDigitalExt.PlacementID == "" {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Missing required parameter 'placement_id'",
+		}}
+	}
```

Comment:
This check is not necessary since you have tagged `placement_id` as a required parameter in your bidder-params file which means PBS core will only call your adapter if this parameter is provided.